homeassistant/components/fritz/services.yaml

```diff
@@ -35,3 +35,31 @@ cleanup:
           integration: fritz
           entity:
             device_class: connectivity
+set_guest_wifi_password:
+  name: Set guest wifi password
+  description: Set a new password for the guest wifi. The password must be between 8 and 63 characters long. If no additional parameter is set, the password will be auto-generated with a length of 32 characters.
+  fields:
+    device_id:
+      name: Fritz!Box Device
+      description: Select the Fritz!Box to check
+      required: true
+      selector:
+        device:
+          integration: fritz
+          entity:
+            device_class: connectivity
+    password:
+      name: Password
+      description: New password for the guest wifi
+      required: false
+      selector:
+        text:
+    length:
+      name: Password length
+      description: Length of the new password. The password will be auto-generated, if no password is set.
+      required: false
+      default: 32
+      selector:
+        number:
+          min: 8
+          max: 63
```

Comment:
Why asking for the length to the end user ?
I would say, use your own password or a default (32bytes long) pwd will be used.
That's it

Reply:
Why not? Its no big deal. I one wants a shorter or longer password but like to rotate it for security reason he could use the built-in mechanism and don't needs to worry about how to generate the password randomly.

Reply:
I'm not a fan of too many options, honestly.
Will ask for a second opinion on that.

Simone

Reply:
Separation of concerns. A service that changes a password to `x` differs from a one that generates a random, time-limited password. For the latter, would it not be necessary to have a way to receive the generated output (which is not possible using service calls)?

I have no idea about the internals of this integration, I just stumbled upon it as I got notified by github, so take my advise with a grain of salt.

Reply:
It's my plan to integrate a camera entity for QR code next. So you could run the password change service and the QR code will be updated. This is my personal use-case.

Reply:
@mib1185 what you think about that ?

Reply:
I'm fine with this optional parameter, but would remove the default of 32, since the fritzconnection lib does already have an own [default value](https://github.com/kbr/fritzconnection/blob/0fb1e2913c4c90428be11676e52e5548cee5bb8b/fritzconnection/lib/fritzwlan.py#L28) implemented for that.
For your @sti0 use case, you can for sure set the `length` to what ever you need.

Reply:
Removed the default value. Now if no value is set the default value of `fritzconnection` will be used (length = 12). I will change that in the docs, too.